sawmill-core/src/main/java/io/logz/sawmill/processors/DocSignatureProcessor.java

```diff
@@ -0,0 +1,154 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorConfigurationException;
+import io.logz.sawmill.exceptions.ProcessorExecutionException;
+import io.logz.sawmill.utilities.JsonUtils;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ProcessorProvider(type = "docSignature", factory = DocSignatureProcessor.Factory.class)
+public class DocSignatureProcessor implements Processor {
+    private final SignatureMode signatureMode;
+    private final LinkedHashSet<String> includeValueFields;
+    private final String signatureFieldName;
+    public DocSignatureProcessor(SignatureMode signatureMode, String signatureFieldName, LinkedHashSet<String> includeValueFields) {
+        this.signatureMode = signatureMode;
+        this.signatureFieldName = signatureFieldName;
+        this.includeValueFields = includeValueFields;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) throws InterruptedException {
+        List<String> signatureCollection;
+        try {
+            signatureCollection = createSignatureCollection(doc);
+        } catch (Exception e) {
+            return ProcessResult.failure(
+                    "failed to create signature, SignatureMode: " + signatureMode,
+                    new ProcessorExecutionException(DocSignatureProcessor.class.getSimpleName(), e));
+        }
+
+        if(signatureCollection.isEmpty())
+            return ProcessResult.failure("signature is empty, SignatureMode: " + signatureMode);
+
+        addSignatureField(doc, signatureCollection);
+        return ProcessResult.success();
+    }
+
+    private List<String> createSignatureCollection(Doc doc) throws InterruptedException {
+        switch(signatureMode) {
+            case FIELDS_VALUES:
+                return getFieldsValues(doc);
+            case FIELDS_NAMES:
+                return new ArrayList<>(extractFieldsNames(doc));
+            case HYBRID:
+                return Stream.concat(getFieldsValues(doc).stream(), extractFieldsNames(doc)
+                        .stream()).collect(Collectors.toList());
+            default: return Collections.emptyList();
+        }
+    }
+
+    private void addSignatureField(Doc doc, List<String> signatureCollection) {
```

Comment:
1] so we agree that it is better to use set hashcode 
2] we want to add a test that {a,b,c} signature is equals to {c,b,a}